src/werkzeug/test.py

```diff
@@ -672,6 +672,7 @@ def get_environ(self):
             input_stream, content_length, boundary = stream_encode_multipart(
                 values, charset=self.charset
             )
+            self.files.add_file("temp", input_stream, filename="temp")
```

Comment:
This relies on the close method which closes all files that are added to it. 

This adds `input_stream` to the files so it gets closed appropriately (which stops the ResourceWarning). Although it works, it doesn't feel too right. Any thoughts on other approaches?
 https://github.com/pallets/werkzeug/blob/2469f13b10b68141b8903a1be2d1c4ed7c86c210/src/werkzeug/test.py#L632